app/services/qa/linked_data/authority_url_service.rb

```diff
@@ -2,59 +2,47 @@
+          action_validation(action)
+          url_config = action_config.url_config
+          selected_substitutions = url_config.extract_substitutions(combined_substitutions(action_config, action, action_request, substitutions, subauthority))
+          Qa::IriTemplateService.build_url(url_config: url_config, substitutions: selected_substitutions)
+        end
 
-        Qa::IriTemplateService.build_url(url_config: url_config, substitutions: selected_substitutions)
-      end
+        private
 
-      def self.action_validation(action)
-        return if [:search, :term].include? action
-        raise Qa::UnsupportedAction, "#{action} Not Supported - Action must be one of the supported actions (e.g. :term, :search)"
-      end
-      private_class_method :action_validation
+          def action_validation(action)
+            return if [:search, :term].include? action
+            raise Qa::UnsupportedAction, "#{action} Not Supported - Action must be one of the supported actions (e.g. :term, :search)"
+          end
 
-      # TODO: elr - rename search and term config methods to be the same to avoid all the ternary checks
-      def self.action_url(auth_config, action)
-        action == :search ? auth_config.url : auth_config.term_url
-      end
-      private_class_method :action_url
+          def combined_substitutions(action_config, action, action_request, substitutions, subauthority)
+            substitutions[action_request_variable(action_config, action)] = action_request
+            substitutions[action_subauth_variable(action_config)] = action_subauth_variable_value(action_config, subauthority) if subauthority.present?
+            substitutions
+          end
 
-      def self.action_request_variable(action_config, action)
-        key = action == :search ? :query : :term_id
-        action == :search ? action_config.qa_replacement_patterns[key] : action_config.term_qa_replacement_patterns[key]
-      end
-      private_class_method :action_request_variable
+          def action_request_variable(action_config, action)
+            key = action == :search ? :query : :term_id
+            action_config.qa_replacement_patterns[key]
+          end
 
-      def self.action_subauth_variable(action_config, action)
-        action == :search ? action_config.qa_replacement_patterns[:subauth] : action_config.term_qa_replacement_patterns[:subauth]
-      end
-      private_class_method :action_subauth_variable
+          def action_subauth_variable(action_config)
+            action_config.qa_replacement_patterns[:subauth]
+          end
 
-      def self.action_subauth_variable_value(action_config, subauthority, action)
-        case action
-        when :search
-          pattern = action_subauth_variable(action_config, action)
-          default = action_config.url_mappings[pattern.to_sym][:default]
-          action_config.subauthorities[subauthority.to_sym] || default
```

Comment:
IriTemplate::UrlConfig and IriTemplate::VariableMap takes care of applying the default, so no need to do that here.